opi/bob/evg/_evg_mxc.bob

```diff
@@ -26,12 +25,8 @@
   <widget type="textentry" version="3.0.0">
     <name>Text Input</name>
     <pv_name>$(P):Mxc$(N)Prescaler-SP</pv_name>
-    <x>90</x>
+    <x>70</x>
     <height>19</height>
-    <foreground_color>
-      <color name="Write_Foreground" red="0" green="255" blue="255">
```

Comment:
It's not the version of Phoebus that matter. Here the problem is that you've removed the macro Write_Foreground, same on different place.

Reply:
It will be something like this.
![image](https://github.com/user-attachments/assets/8657fd3a-e7f4-4a54-a1d8-60c56519e200)
I guess I need to change the default background of the text input and text update.
Do you prefer the following style?
![image](https://github.com/user-attachments/assets/203c4caa-3a31-4ba1-b479-7f82ba854871)

Do you have a specific background color for the text update and text entry? I am OK with any color as long as the contrast between the foreground and background is enough.

Thanks!

Ran Hong

Reply:
I think we don't understand each other. 
I don't care about the color, but I care about the respect of the color name macro.

As an example I use the file color.def there : https://github.com/epics-modules/mrfioc2/blob/master/opi/bob/color.def

If you don't know the color.def file, it's just a file where you define the code rgb of a macro. The advantage is that from a project to another if we have a different color convention, you just have to change your own color.def but not the bob files.
(the same exist with the font, but I don't use one)

In phoebus configuration file you can define which files to use there : 
 
```
# Display configuration
org.csstudio.display.builder.model/color_files=color.def
org.csstudio.display.builder.model/font_files=font.def
org.phoebus.ui/welcome=welcome.html
```

I have tried in the color.def to be as close as possible as the standard name to be a maximum compliant with everyone

Reply:
HI @agaget ,

Indeed I didn't know that the default color can be configured in this way. I used the color.def from the repository and updated the foreground colors of the read/write fields. 

Could you take a look at the commit that I just pushed?

Thanks for your instructions on defining the color macros.

Ran Hong

Reply:
BTW, the reason why I removed the macro Write_Foreground in the first place was that I planned to use the default color for the text entry, which is the macro "Text". My default color.def has different colors for read and write background colors.

I like the idea to have Write_Foreground and Read_Foreground, so people have more flexibility to make their own color schemes using color.def.